### PR TITLE
Remove aws-xray-sdk

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

- Remove `aws-xray-sdk` from the layer
- Refactors how we create the dummy metadata datadog subsegment

### Motivation

Cold start performance improvements

### Testing Guidelines

Unit tests + integration tests + manual test

Screen Shot 2021-09-03 at 1 53 42 PM
Screen Shot 2021-09-03 at 1 53 36 PM
Screen Shot 2021-09-03 at 1 53 31 PM

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [x] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: LICENSE-3rdparty.csv [hunk 1]
```diff
@@ -1,6 +1,4 @@
 Component,Origin,License,Copyright
-aws-xray-sdk-python,github.com/aws/aws-xray-sdk-python,Apache-2.0,
 flake8,gitlab.com/pycqa/flake8,MIT,"Copyright (C) 2011-2013 Tarek Ziade <tarek@ziade.org>. Copyright (C) 2012-2016 Ian Cordasco <graffatcolmingov@gmail.com>."
 nose2,github.com/nose-devs/nose2,BSD-2-Clause,"Copyright (c) 2012, Jason Pellerin. All rights reserved."
-requests,github.com/kennethreitz/requests,Apache-2.0,"Copyright 2018 Kenneth Reitz"
```
Comment (maxday) [inline LICENSE-3rdparty.csv:5]: requests was previously removed

File: datadog_lambda/xray.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+def build_segment(context, key, metadata):
+
+    segment = json.dumps(
+        {
+            "id": generate_random_id(),
+            "trace_id": context["trace_id"],
+            "parent_id": context["parent_id"],
+            "name": XraySubsegment.NAME,
+            "start_time": time.time(),
```
Comment (DarcyRaynerDD) [inline datadog_lambda/xray.py:84]: Just want to verify that time.time() returns in the correct unit: https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html . I believe there might be a bug in the node version where this timestamp is off by an order of magnitude.

**Review by DarcyRaynerDD — APPROVED** [review]
LGTM, left one thing we need to check

File: datadog_lambda/xray.py (hunk 2 above)
Comment (maxday) [reply datadog_lambda/xray.py:84]: I've check at https://github.com/aws/aws-xray-sdk-python/blob/894b419518e14b5cb2a75293ed4da17f87b2a7c0/aws_xray_sdk/core/models/entity.py#L39 and it seems correct?

**Pull request merged**